pubsub/streaming-analytics/PubSubToGCS_test.py

```diff
@@ -12,89 +12,68 @@
-
-
-@pytest.fixture
-def topic_path(publisher_client):
-    topic_path = publisher_client.topic_path(PROJECT, TOPIC)
-
-    try:
-        publisher_client.delete_topic(topic_path)
-    except Exception:
-        pass
-
-    response = publisher_client.create_topic(topic_path)
-    yield response.name
-
-
-def _infinite_publish_job(publisher_client, topic_path):
-    while True:
-        future = publisher_client.publish(
-            topic_path, data='Hello World!'.encode('utf-8'))
-        future.result()
-        time.sleep(10)
-
-
-def test_run(publisher_client, topic_path):
-    """This is an integration test that runs `PubSubToGCS.py` in its entirety.
-    It checks for output files on GCS.
-    """
-
-    # Use one process to publish messages to a topic.
-    publish_process = mp.Process(
-        target=lambda: _infinite_publish_job(publisher_client, topic_path))
-
-    # Use another process to run the streaming pipeline that should write one
-    # file to GCS every minute (according to the default window size).
-    pipeline_process = mp.Process(
-        target=lambda: sp.call([
-            'python', 'PubSubToGCS.py',
-            '--project', PROJECT,
-            '--runner', 'DirectRunner',
-            '--temp_location', tempfile.mkdtemp(),
-            '--input_topic', topic_path,
-            '--output_path', 'gs://{}/pubsub/{}/output'.format(BUCKET, UUID),
-        ])
-    )
-
-    publish_process.start()
-    pipeline_process.start()
-
-    # Times out the streaming pipeline after 90 seconds.
-    pipeline_process.join(timeout=90)
-    # Immediately kills the publish process after the pipeline shuts down.
-    publish_process.join(timeout=0)
-
-    pipeline_process.terminate()
-    publish_process.terminate()
-
-    # Check for output files on GCS.
-    gcs_client = beam.io.gcp.gcsio.GcsIO()
-    # This returns a dictionary.
-    files = gcs_client.list_prefix('gs://{}/pubsub/{}'.format(BUCKET, UUID))
-    assert len(files) > 0
-
-    # Clean up. Delete topic. Delete files.
-    publisher_client.delete_topic(topic_path)
-    gcs_client.delete_batch(list(files))
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.testing.test_pipeline import TestPipeline
+from apache_beam.testing.test_stream import TestStream
+from apache_beam.testing.test_utils import TempDir
+from apache_beam.transforms.window import TimestampedValue
+
+import PubSubToGCS
+
+PROJECT = os.environ["GCLOUD_PROJECT"]
+BUCKET = os.environ["CLOUD_STORAGE_BUCKET"]
+UUID = uuid.uuid1().hex
+
+
+class PubSubToGCSTest(unittest.TestCase):
+    def test_pubsub_to_gcs(self):
+
+        pipeline_options = PipelineOptions(
+            project=PROJECT,
+            runner="DirectRunner",
+            temp_location=TempDir().get_path(),
+            streaming=True,
+            save_main_session=True,
+        )
+
+        with TestPipeline(options=pipeline_options) as p:
+
+            mocked_pubsub_stream = (
```

Comment:
I'm not sure I like this style. Did you run this through black?

Reply:
I left all formatting to `black` actually. How do you usually auto-format code?

Reply:
@GoogleCloudPlatform/python-samples-owners Should we back out black from the noxfile/lint check while we're discussing changes to the style guide?

Reply:
I'm pretty sure I prefer our originating style over black but if I'm the only holdout here I'm open to revising our style.

Reply:
@gguuss What do you use to auto-format code that passes lint? I have touched so many files by now.

Reply:
YAPF can format to Google style but I don't use it and instead write Python by hand.

Reply:
YAPF introduces weird formatting too.